go-controller/pkg/ovn/namespace.go

```diff
@@ -0,0 +1,122 @@
+package ovn
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	kapi "k8s.io/client-go/pkg/api/v1"
+	"sync"
+	"time"
+)
+
+func (oc *Controller) waitForNamespaceEvent(namespace string) {
+	for {
+		if oc.namespacePolicies[namespace] != nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+}
+
+func (oc *Controller) addPodToNamespaceAddressSet(ns, address string) {
+	if oc.namespacePolicies[ns] == nil {
+		return
+	}
+
+	oc.namespaceMutex[ns].Lock()
+	defer oc.namespaceMutex[ns].Unlock()
+
+	oc.namespaceAddressSet[ns] = append(oc.namespaceAddressSet[ns],
+		address)
+	oc.setAddressSet(ns, oc.namespaceAddressSet[ns])
+}
+
+func (oc *Controller) deletePodFromNamespaceAddressSet(ns, address string) {
+	if address == "" || oc.namespacePolicies[ns] == nil {
+		return
+	}
+
+	oc.namespaceMutex[ns].Lock()
+	defer oc.namespaceMutex[ns].Unlock()
+
+	for i, item := range oc.namespaceAddressSet[ns] {
+		if item == address {
+			oc.namespaceAddressSet[ns] = append(
+				oc.namespaceAddressSet[ns][:i],
+				oc.namespaceAddressSet[ns][i+1:]...)
+		}
+	}
+
+	oc.setAddressSet(ns, oc.namespaceAddressSet[ns])
+}
+
+func (oc *Controller) addNamespace(ns *kapi.Namespace) {
```

Comment:
@rajatchopra 
Can you expand on your suggestion to avoid the waitForNamespaceEvent function? Are you suggesting that when we don't see a namespace in network policy event, we just make a call to API server to get a namespace?